### PR TITLE
Adds lungs internal organ

### PR DESCRIPTION
**Pull request by Core0verload**
Lungs are added to humans and monkeys. They go into the chest and work like the rest of organs. When removed, they cause constant oxyloss and muteness.

"Revive" restores lungs in mobs that had their lungs removed.

I didn't added lungs to aliens because, AFAIK, they don't have any lungs and rely on some sort of completely non-human breathing process involving the tubes on their backs. I don't have a sprite for this, and adding it as organ would require adding on-mob organ icon system - i just don't want to mess with human's icon update without a really good reason.

:cl: CoreOverload
rscadd: A new organ - lungs. When removed, it will give you breathing and speaking troubles. 
/:cl:

### CODE REVIEW

**Comment by TechnoAlchemisto** [comment]
Is this part of a larger project or is this just adding it to add it?

**Comment by Core0verload** [comment]
It's both. The lungs and some of the cyberimplants i'm currently adding in other PRs were planned as a part of my "Organ Update 3" that never happened.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1078,8 +1078,12 @@
 	if((H.status_flags & GODMODE))
 		return
 
-	if(!breath || (breath.total_moles() == 0))
-		if(H.reagents.has_reagent("epinephrine"))
+	var/lungs = H.getorganslot("lungs")
+	if(!lungs)
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species.dm]: what about NOBREATH? a species that doesn't breathe shouldn't suffer from missing lungs.

Comment (Core0verload) [inline code/modules/mob/living/carbon/human/species.dm]: A species that doesn't breathe shouldn't have the lungs in a first place, but i'm too lazy to make different organ sets for each subspecies possible. At least for now.

Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species.dm]: Yes, but if those species didn't have lungs they'd take the oxy damage from this.
this is my point.

**Comment by Fox-McCloud** [comment]
Very nice :+1:

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -100,9 +100,13 @@
 	if((status_flags & GODMODE))
 		return
 
+	var/lungs = getorganslot("lungs")
+	if(!lungs)
+		adjustOxyLoss(2)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/life.dm]: Won't this mean that Xenos are going to take damage, or do they have an override somewhere?

**Comment by phil235** [comment]
>"Revive" restores lungs in mobs that had their lungs removed.

Why not just fail the revival if they don't have lungs? (just like missing brain)

**Comment by Core0verload** [comment]
The reason why revive fails when there is no brain in body is that no brain == no client. It's easy to create new brain, but revived clientless body would be useless for most of the revive use cases.

File: code/modules/mob/living/carbon/life.dm (hunk 2 above)
Comment (Core0verload) [inline code/modules/mob/living/carbon/life.dm]: They have an override.

**Comment by phil235** [comment]
Before my very recent refactor of revive, it did not fail even if there was no client or brain. In the absence of a brain organ, the mob was revived and died immediately afterwards.

Keep in mind that revive() isn't only used by admin procs, it's called by defib, strange reagent, healing staff, etc... Do we really want to be able to produce an infinite amount of lungs with a defibrillator?

**Comment by Core0verload** [comment]
Defib uses revive without a `full_heal=1` option. Same for strange reagent. And lungs regenerate in "full healing" code. 

As for healing staff/wand - it's literally magic. All code segments that use `full_heal=1` on humans/monkeys: 

- Healing staff and wand (magic)
- Wish Granter immortality (magic)
- Die of Fate roll 12 (magic)
- Blue Orb (magic)
- Necromantic stone (magic)
- Shadowling thrall revive (dark magic)
- Admin panel Heal (badmin magic)
- Changeling regeneration (surprisingly, not magic, but can be justified anyway)

**Comment by ChangelingRain** [comment]
Lungs also regenerate with full_heal=0 though?

**Comment by Core0verload** [comment]
They **DO NOT** regenerate without `full_heal = 1` because they are restored in `fully_heal` proc, which is called only by `revive(full_heal = 1)`.

**Pull request merged**